Revise test_remote_backend_properties_filter_date

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### What is the current behavior?

A couple of issues arose from the test included #277:
* the test is using `.fromisoformat()`, which is [only available in 3.7+](https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat) - can we find an alternative way that uses only Pyhon 3.5 constructs?
* the comparison between `properties.last_update_date` and the filter might fail due to time zone issues:

```
    def test_remote_backend_properties_filter_date(self):
        """Test backend properties filtered by date."""
        backends = self.provider.backends(simulator=False)
    
        datetime_filter = datetime.fromisoformat('2019-02-01T00:00:00.000')
        for backend in backends:
            with self.subTest(backend=backend):
                properties = backend.properties(datetime=datetime_filter)
                if isinstance(properties, BackendProperties):
>                   self.assertLessEqual(properties.last_update_date, datetime_filter)
E                   TypeError: can't compare offset-naive and offset-aware datetimes
```



### Steps to reproduce the problem



### What is the expected behavior?



### Suggested solutions

--- a/qiskit/providers/ibmq/ibmqbackend.py
+++ b/qiskit/providers/ibmq/ibmqbackend.py
@@ -98,6 +98,7 @@ class IBMQBackend(BaseBackend):
                 warnings.warn('Retrieving the properties of a '
                               'backend in a specific datetime is '
                               'only available when using IBM Q v2')
+                return None
 
             # Do not use cache for specific datetime properties.
             api_properties = self._api.backend_properties(self.name(), datetime=datetime)

--- a/qiskit/providers/ibmq/ibmqbackend.py
+++ b/qiskit/providers/ibmq/ibmqbackend.py
@@ -94,6 +94,11 @@ class IBMQBackend(BaseBackend):
         """
         # pylint: disable=arguments-differ
         if datetime:
+            if not isinstance(self._api, BaseClient):
+                warnings.warn('Retrieving the properties of a '
+                              'backend in a specific datetime is '
+                              'only available when using IBM Q v2')
+
             # Do not use cache for specific datetime properties.
             api_properties = self._api.backend_properties(self.name(), datetime=datetime)
             if not api_properties:

--- a/test/ibmq/test_ibmq_provider.py
+++ b/test/ibmq/test_ibmq_provider.py
@@ -96,12 +96,13 @@ class TestIBMQProvider(IBMQTestCase, providers.ProviderTestCase):
         """Test backend properties filtered by date."""
         backends = self.provider.backends(simulator=False)
 
-        datetime_filter = datetime.fromisoformat('2019-02-01T00:00:00.000')
+        datetime_filter = datetime(2019, 2, 1).replace(tzinfo=None)
         for backend in backends:
             with self.subTest(backend=backend):
                 properties = backend.properties(datetime=datetime_filter)
                 if isinstance(properties, BackendProperties):
-                    self.assertLessEqual(properties.last_update_date, datetime_filter)
+                    last_update_date = properties.last_update_date.replace(tzinfo=None)
+                    self.assertLessEqual(last_update_date, datetime_filter)
                 else:
                     self.assertEqual(properties, None)
 

--- a/test/ibmq/test_ibmq_provider.py
+++ b/test/ibmq/test_ibmq_provider.py
@@ -92,20 +92,6 @@ class TestIBMQProvider(IBMQTestCase, providers.ProviderTestCase):
             if backend.configuration().simulator:
                 self.assertEqual(properties, None)
 
-    def test_remote_backend_properties_filter_date(self):
-        """Test backend properties filtered by date."""
-        backends = self.provider.backends(simulator=False)
-
-        datetime_filter = datetime(2019, 2, 1).replace(tzinfo=None)
-        for backend in backends:
-            with self.subTest(backend=backend):
-                properties = backend.properties(datetime=datetime_filter)
-                if isinstance(properties, BackendProperties):
-                    last_update_date = properties.last_update_date.replace(tzinfo=None)
-                    self.assertLessEqual(last_update_date, datetime_filter)
-                else:
-                    self.assertEqual(properties, None)
-
     def test_remote_backend_defaults(self):
         """Test backend pulse defaults."""
         remotes = self.provider.backends(simulator=False)
@@ -190,3 +176,17 @@ class TestAccountProvider(TestIBMQProvider):
         # pylint: disable=arguments-differ
         ibmq = IBMQFactory()
         return ibmq.enable_account(qe_token, qe_url)
+
+    def test_remote_backend_properties_filter_date(self):
+        """Test backend properties filtered by date."""
+        backends = self.provider.backends(simulator=False)
+
+        datetime_filter = datetime(2019, 2, 1).replace(tzinfo=None)
+        for backend in backends:
+            with self.subTest(backend=backend):
+                properties = backend.properties(datetime=datetime_filter)
+                if isinstance(properties, BackendProperties):
+                    last_update_date = properties.last_update_date.replace(tzinfo=None)
+                    self.assertLessEqual(last_update_date, datetime_filter)
+                else:
+                    self.assertEqual(properties, None)

